src/data/zsqrtd/basic.lean

```diff
@@ -289,6 +290,12 @@ def norm (n : ℤ√d) : ℤ := n.re * n.re - d * n.im * n.im
 @[simp] lemma norm_mul (n m : ℤ√d) : norm (n * m) = norm n * norm m :=
 by { simp only [norm, mul_im, mul_re], ring }
 
+/-- `norm` as a `monoid_hom`. -/
+def norm_monoid_hom : ℤ√d →* ℤ :=
```

Comment:
This kind of duplicates the stuff on line 309 - we should probably write one in terms of the other